lstchain/reco/r0_to_dl1.py

```diff
@@ -838,3 +767,101 @@ def rescale_dl1_charge(event, scaling_factor):
 
     for tel_id, tel in event.dl1.tel.items():
         tel.image *= scaling_factor
+
+
+def r0_dl1_muon_analysis(event, telescope_id, r1_dl1_calibrator_for_muon_rings, subarray, muon_parameters, dragon_time):
+    """
+    Muon analysis in the stage r0 to dl1.
+    Fills the `muon_parameters` table.
+
+    Parameters
+    ----------
+    event: `ctapipe.containers.DataContainer`
+    telescope_id: int
+    r1_dl1_calibrator_for_muon_rings: `LSTCameraCalibrator`
+    subarray: `ctapipe.instrument.subarray.SubarrayDescription`
+    muon_parameters: dict `lstchain.image.muon.create_muon_table`
+    dragon_time: `astropy.Quantity`
+
+    """
+    # minimum number of pe in a pixel to include it
+    # in calculation of muon ring time (peak sample):
+    min_pe_for_muon_t_calc = 10.
+
+    # FIXME: no need to read telescope characteristics like foclen for every event!
+    foclen = subarray.tel[telescope_id].optics.equivalent_focal_length
+    mirror_area = subarray.tel[telescope_id].optics.mirror_area
+
+    bad_pixels = event.mon.tel[telescope_id].calibration.unusable_pixels[0]
+    # Set to 0 unreliable pixels:
+    image = event.dl1.tel[telescope_id].image*(~bad_pixels)
```

Comment:
Is there in ctapipe already some pixel interpolator? Would be interesting to test it. Should be better in case of isolated bad pixels (perhaps a bit dangerous when a whole pixel cluster is unreliable).

Reply:
I don't think there is, but yes, I think it is necessary.

Reply:
It should be easy enough to implement approaches like a mean of correct neighbors using the neighbor matrix.